its/it/azdo.ts

```diff
@@ -9,6 +22,46 @@ export function getAzdoApi(): vm.WebApi {
   return new vm.WebApi(AZDO_BASE_URL + AZDO_ORGANIZATION, credentialHandler);
 }
 
+export async function getTemplatePipeline(buildApi: IBuildApi): Promise<BuildDefinition> {
+  // Find an existing pipeline to use as a template
+  const definitions = await buildApi.getDefinitions(AZDO_PROJECT);
+  const existingDefinition = definitions.find((d) => d.name?.startsWith(AZDO_PIPELINE_NAME_PREFIX));
+  if (!existingDefinition) {
+    throw new Error("No existing pipeline found to use as a template");
+  }
+  // Get full object
+  const existingDefinitionFull = await buildApi.getDefinition(
+    AZDO_PROJECT,
+    existingDefinition.id as number,
+  );
+  return existingDefinitionFull;
+}
+
+export async function createPipeline(
+  buildApi: IBuildApi,
+  template: BuildDefinition,
+  pipelineName: string,
+): Promise<void> {
+  const existingPipelines = await buildApi.getDefinitions(AZDO_PROJECT, pipelineName);
+  if (existingPipelines.length > 0) {
+    console.log(`Pipeline ${pipelineName} already exists. Skipping creation.`);
+    return;
+  }
+
+  console.log(`Creating pipeline ${pipelineName}`);
+  await buildApi.createDefinition(
+    {
+      ...template,
+      process: {
+        ...template.process,
+        yamlFilename: path.join(FIXTURES_PATH, pipelineName + ".yml"),
+      } as unknown as any,
+      name: pipelineName,
+    },
+    AZDO_PROJECT,
```

Comment:
how can we be certain the template we are copying is good enough? I know we are creating the template now from the `generate-it` task, but what if we change something in the base? we would need to clean up all existing templates to avoid a clash right?

Reply:
Maybe it's better to create them from scratch. I'll restart the investigation tomorrow. The solution will be a bit more complex, but at least we know we won't have side effects or unforeseen consequences as you call out

Reply:
what about ALWAYS generating a "template.yml" that we can always reference and always generate? and just use that as the template? instead of relying on a random one

Reply:
We can actually use the main pipeline `SonarSource.sonar-scanner-azdo` as a template directly!